core/kazoo_bindings-1.0.0/src/kazoo_bindings.erl

```diff
@@ -676,53 +676,79 @@ map_processor(Routing, Payload, Bindings) when not is_list(Payload) ->
     map_processor(Routing, [Payload], Bindings);
 map_processor(Routing, Payload, Bindings) ->
     RoutingParts = lists:reverse(binary:split(Routing, <<".">>, ['global'])),
-    Map = fun(Responder) ->
-                  apply_responder(Responder, Payload)
-          end,
+    Map = map_responder_fun(Payload),
 
-    lists:foldl(fun(#kz_binding{binding=Binding
-                                ,binding_responders=Responders
-                               }
-                    ,Acc
-                   ) when Binding =:= Routing ->
-                        lager:debug("exact match for ~s", [Routing]),
-                        [catch Map(Responder)
-                         || Responder <- queue:to_list(Responders)
-                        ] ++ Acc;
-                   (#kz_binding{binding_parts=BParts
-                                ,binding_responders=Responders
-                               }
-                    ,Acc
-                   ) ->
-                        case matches(BParts, RoutingParts) of
-                            'false' -> Acc;
-                            'true' ->
-                                lager:debug("matched ~p to ~p", [BParts, RoutingParts]),
-                                [catch Map(Responder)
-                                 || Responder <- queue:to_list(Responders)
-                                ] ++ Acc
-                        end
+    lists:foldl(fun(Binding, Acc) ->
+                        map_processor_fold(Binding, Acc, Map, Routing, RoutingParts)
                 end
                 ,[]
                 ,Bindings
                ).
 
+-type map_responder_fun() :: fun((kz_responder()) -> any()).
+-spec map_responder_fun(payload()) -> map_responder_fun().
+map_responder_fun(Payload) ->
+    fun(Responder) ->
+            apply_responder(Responder, Payload)
+    end.
```

Comment:
I would counter that this seems like a good way to hide some clutter from the call site that would distract from reading `map_processor/3`